spectrum-network/src/peer_conn_handler.rs

```diff
@@ -218,6 +220,12 @@ impl IntoConnectionHandler for PartialPeerConnHandler {
                 )
             })
         }));
+
+        #[cfg(not(feature = "test_peer_punish_too_slow"))]
```

Comment:
To test peer punishment for being too slow we need to throttle the receiver so that the buffer reaches its limit which will lead us to send `StreamNotification::ForceClose`. The problem is that we don't have control over the creation of `PeerConnHandler` instances; they are currently created through `NetworkBehaviour::new_handler(&mut self)`. A way around it is to conditionally compile; if the feature "test_peer_punish_too_slow" is set, the throttling occurs.